x-pack/plugins/alerting/server/application/rule/methods/find/schemas/find_rules_schemas.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+
+export const findRulesOptionsSchema = schema.object(
+  {
+    perPage: schema.maybe(schema.number()),
+    page: schema.maybe(schema.number()),
+    search: schema.maybe(schema.string()),
+    defaultSearchOperator: schema.maybe(
+      schema.oneOf([schema.literal('AND'), schema.literal('OR')])
+    ),
+    searchFields: schema.maybe(schema.arrayOf(schema.string())),
+    sortField: schema.maybe(schema.string()),
+    sortOrder: schema.maybe(schema.oneOf([schema.literal('asc'), schema.literal('desc')])),
+    hasReference: schema.maybe(
+      schema.object({
+        type: schema.string(),
+        id: schema.string(),
+      })
+    ),
+    fields: schema.maybe(schema.arrayOf(schema.string())),
+    filter: schema.maybe(
+      schema.oneOf([schema.string(), schema.recordOf(schema.string(), schema.any())])
+    ),
+    filterConsumers: schema.maybe(schema.arrayOf(schema.string())),
+  },
+  { unknowns: 'allow' }
```

Comment:
What is the reason of allowing `unknowns`?

Reply:
security solution is passing through some props into this method that is passed through to the saved object client. These props are not listed as supported types to the `find` function. I dont think the props are doing anything but I was afraid to remove it.

Reply:
>These props are not listed as supported types to the find function

But they use it anyway? How TS does not throw an error? Did we support it in the first place?

Reply:
Well it doesn't cause errors because we pull off all of the fields that the method uses. So extra props are just discarded. However we now validate the params so without passthrough, it starts to throw.

Reply:
I see. Thanks!